aten/src/ATen/native/_scaled_mm.cpp

```diff
@@ -0,0 +1,189 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/core/Tensor.h>
+#include <ATen/Config.h>
+#include <ATen/Context.h>
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/Resize.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/_scaled_mm_native.h>
+#include <ATen/ops/empty.h>
+#endif
+
+#if !AT_MKLDNN_ENABLED()
+namespace at::native {
+
+Tensor&
+_scaled_mm_out_cpu(const Tensor& mat1, const Tensor& mat2,
+          const Tensor& scale_a,
+          const Tensor& scale_b,
+          const std::optional<at::Tensor>& bias,
+          const std::optional<at::Tensor>& scale_result,
+          std::optional<c10::ScalarType> out_dtype,
+          bool use_fast_accum,
+          Tensor& out) {
+  TORCH_INTERNAL_ASSERT(false, __func__, ": ATen not compiled with MKLDNN support");
+}
+
+Tensor
+_scaled_mm_cpu(const Tensor& mat_a, const Tensor& mat_b,
+          const Tensor& scale_a,
+          const Tensor& scale_b,
+          const std::optional<at::Tensor>& bias,
+          const std::optional<at::Tensor>& scale_result,
+          std::optional<c10::ScalarType> out_dtype,
+          bool use_fast_accum) {
+  TORCH_INTERNAL_ASSERT(false, __func__, ": ATen not compiled with MKLDNN support");
+}
+}
+
+#else // AT_MKLDNN_ENABLED
+namespace at::native {
+
+Tensor&
+_scaled_mm_out_cpu(const Tensor& mat1, const Tensor& mat2,
```

Comment:
Put onednn related binding code in aten/src/Aten/native/mkldnn/Linear.cpp and name it `mkldnn_scaled_mm` or something that is reasonable.

this function should be:
```
if mkldnn_is_compiled:
  mkldnn_scaled_mm(...)
else // mkldnn is not compiled
  mkldnn_scaled_mm() {
  TORCH_CHECK(false, "mkldnn is not compiled");
}
```

and then let _scaled_mm_out_cpu call mkldnn_scaled_mm

Reply:
Done.